ToolChain/Assembler/Parser/Parser.py

```diff
@@ -116,37 +124,46 @@ def _parseCodeFile(self):
         """
         This method will take the content of the file and build a list of instruction with it.
         That list is in the same order as what is in the actual file content.
-        :return: list: this returns a list of instruction that are present in the file.
+        :return: list: this returns a list of the instructions that are present in the file.
         """
-        instructionList = []  # Will be built as [binaryInstruction, ...]
-        memoryReferenceDict = {}  # Will be built as [{"ref": relAddress}, ...]
-        relativeAddressCounter = 0
-        lineNo = 0
+
+        instructionList = []            # Will contain the list of binary instructions
+        memoryReferenceDict = {}        # Contains memory references. Will contain label along with its relative address
+        relativeAddressCounter = 0      # Counter cumulatively adds length of instructions.
+        lineNo = 0                      # Keeps track of current line being evaluated to print errors at specific lines
 
         for line in self.fileContent:
+
             lineNo += 1
+
             try:
                 sline = line.split()
                 if len(sline) > 0:
-                    # We are here, we have a code line
-                    buildInstruction, lineInstruction = self._parseCodeLine(sline)
+                    # If we make it here, there was something on the line
+                    # We move to the next step and parse the line to get the appropriate code for what's written
+
+                    buildInstruction, lineInstruction = self._parseCodeLine(line)
```

Comment:
Why change this to sending a line?

By sending a line, you are forcing each function to re-split the line. Not that it's a crazy expensive operation but it doesn't make sense to split it at line 140, verify the length, then discard the result to regenerate it later.